lib/srv/authhandlers.go

```diff
@@ -198,9 +205,14 @@ func (h *AuthHandlers) UserKeyAuth(conn ssh.ConnMetadata, key ssh.PublicKey) (*s
 	// Check that the user certificate uses supported public key algorithms, was
 	// issued by Teleport, and check the certificate metadata (principals,
 	// timestamp, etc). Fallback to keys is not supported.
+	clock := h.Clock.Now
+	if clock == nil {
```

Comment:
`clock` here is a method value.
IIUC, it will never be nil. But calling it will call the `Now` method with a nil receiver.

to be safer, invert the check:
```
clock := time.Now
if h.Clock != nil {
  clock = h.Clock.Now
}
```

Reply:
Yes, I realized that later myself - thanks for pointing it out.